src/comm/Device.cc

```diff
@@ -82,10 +82,6 @@ Device& global_device()
         }
     }
 
-#if CELERITAS_USE_CUDA && defined(_OPENMP)
-    CELER_NOT_IMPLEMENTED("OpenMP support with CUDA");
-#endif
```

Comment:
I added this to protect against using CUDA simultaneously with OpenMP because `global_device` is a singleton and doesn't support using one GPU per thread (which is the likely mechanism for Geant4 integration).

Reply:
Understood -- clearly we don't want this called from multiple threads. But leaving it in prevents compiling altogether unless we assume that OpenMP is only going to be used with CELERITAS_USE_CUDA turned off.

Reply:
Good point, so maybe we need a global "CPU Threads" state (analogous to the Device and Communicator classes) that can store the anticipated number of threads, since I don't think OpenMP lets you query the planned parallelism outside of a parallel thread block. In other words, this would be nice but I don't think it'll work:
```c++
int global_threads()
{
#ifdef _OPENMP
   return omp_num_threads();
#else
   return 1;
#endif
}
```
and so we can't say
```c++
if (device && num_threads()) > 1)
{
   CELER_NOT_IMPLEMENTED("OpenMP support with CUDA");
}
```

I think @pcanal was the one who suggested this code since we will probably using thread-parallel Geant4.

Reply:
The issue was (still is, I think) that (some of) the (cuda) code in Device.cc is not thread safe (nor supporting multi device).  So this was a shortcut to remind us to improve the code (and/or the protection against misuse) if we ever mix the two.